src/array_api_extra/_funcs.py

```diff
@@ -46,3 +47,48 @@ def atleast_nd(x: Array, *, ndim: int, xp: ModuleType) -> Array:
         x = xp.expand_dims(x, axis=0)
         x = atleast_nd(x, ndim=ndim, xp=xp)
     return x
+
+
+def cov(x: Array, *, xp: ModuleType) -> Array:
+    """..."""
+    x = xp.asarray(x, copy=True)
+    dtype = (
+        xp.float64 if xp.isdtype(x.dtype, "integral") else xp.result_type(x, xp.float64)
+    )
+
+    x = atleast_nd(x, ndim=2, xp=xp)
+    x = xp.astype(x, dtype)
+
+    avg = mean(x, axis=1, xp=xp)
+    fact = x.shape[1] - 1
+
+    if fact <= 0:
+        warnings.warn("Degrees of freedom <= 0 for slice", RuntimeWarning, stacklevel=2)
+        fact = 0.0
+
+    x -= avg[:, None]
+    y_transpose = x.T
+    if xp.isdtype(y_transpose.dtype, "complex floating"):
+        y_transpose = xp.conj(y_transpose)
+    c = x @ y_transpose
+    c /= fact
+    axes = tuple(axis for axis, length in enumerate(c.shape) if length == 1)
+    return xp.squeeze(c, axis=axes)
+
+
+def mean(
+    x: Array,
+    /,
+    *,
+    axis: int | tuple[int, ...] | None = None,
+    keepdims: bool = False,
+    xp: ModuleType,
+) -> Array:
+    """..."""
+    if xp.isdtype(x.dtype, "complex floating"):
```

Comment:
@lucascolley As the mean is well-defined for complex numbers, I suggest opening an issue directly on the Array API issue tracker. This is likely something that we can add to the 2024 revision of the standard, provided there is broad enough support.

Reply:
This is also an unfortunate instance where `np.mean(x)` is about 2x faster than this implementation for complex `x` (and presumably other eager libraries will be similar).